tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/stream/BStreamValueTest.java

```diff
@@ -240,21 +243,30 @@ public void testStreamPublishingAndSubscriptionForAssignableUnionTypeStream() {
         Assert.assertEquals(receivedEvents.size(), publishedEvents.size(), "Number of Events received does not "
                 + "match the number published");
         for (int i = 0; i < publishedEvents.size() - 1; i++) {
-            Assert.assertEquals(receivedEvents.getRefValue(i), publishedEvents.getRefValue(i),
-                                "Received event does not match the published event");
+            Assert.assertTrue(BVM.isEqual(receivedEvents.getRefValue(i), publishedEvents.getRefValue(i),
```

Comment:
Since the BVM will be deprecated, and we can't use BValues with the TypeChecker, shall we update the test to check equality on the Ballerina side and return a `boolean` instead?

Reply:
fixed